test/server/cards/01_SOR/events/DontGetCocky.spec.ts

```diff
@@ -0,0 +1,104 @@
+
+                context.player1.clickPrompt('Reveal another card');
+                expect(context.getChatLogs(1)).toContain('player1 reveals Wilderness Fighter due to Don\'t Get Cocky');
+                expect(context.player1).toHaveEnabledPromptButtons(['Reveal another card', 'Stop revealing cards']);
+
+                context.player1.clickPrompt('Reveal another card');
+                expect(context.getChatLogs(1)).toContain('player1 reveals Consortium StarViper due to Don\'t Get Cocky');
+                expect(context.player1).toHaveEnabledPromptButtons(['Reveal another card', 'Stop revealing cards']);
+
+                context.player1.clickPrompt('Stop revealing cards');
+                expect(context.atatSuppressor.damage).toBe(7);
+                expect([context.allianceDispatcher, context.medalCeremony, context.wildernessFighter, context.consortiumStarviper]).toAllBeInBottomOfDeck(context.player1, 4);
+                expect(context.player1Object.getTopCardOfDeck()).toBe(context.entrenched);
+
+                reset();
+
+                // reveal 5 cards, do no damage because the total cost is too high
+                context.player1.clickCard(context.dontGetCocky);
+                context.player1.clickCard(context.atatSuppressor);
+
+                context.player1.clickPrompt('Reveal another card');
+                context.player1.clickPrompt('Reveal another card');
+                context.player1.clickPrompt('Reveal another card');
+                context.player1.clickPrompt('Reveal another card');
+                context.player1.clickPrompt('Stop revealing cards');
+                expect(context.atatSuppressor.damage).toBe(0);
+                expect([context.allianceDispatcher, context.medalCeremony, context.wildernessFighter, context.consortiumStarviper, context.entrenched]).toAllBeInBottomOfDeck(context.player1, 5);
+                expect(context.player1Object.getTopCardOfDeck()).toBe(context.karabast);
+
+                reset();
+
+                // the controller can continue revealing cards regardless of total cost so far, up to 7 cards maximum
+                context.player1.clickCard(context.dontGetCocky);
+                context.player1.clickCard(context.atatSuppressor);
+
+                context.player1.clickPrompt('Reveal another card');
+                context.player1.clickPrompt('Reveal another card');
+                context.player1.clickPrompt('Reveal another card');
+                context.player1.clickPrompt('Reveal another card');
+                context.player1.clickPrompt('Reveal another card');
+                context.player1.clickPrompt('Reveal another card');
+                // 7 cards revealed, no choices left
+                expect(context.player2).toBeActivePlayer();
+                expect(context.atatSuppressor.damage).toBe(0);
+                expect(context.player1Object.getTopCardOfDeck()).toBe(context.swoopRacer);
```

Comment:
These are great! I think we need just one more to test what happens if there aren't enough cards in deck